src/ScriptBuilder/Writers/TimeoutWriter.cs

```diff
@@ -1,21 +1,41 @@
+using System;
 using System.IO;
 using NServiceBus.Persistence.Sql.ScriptBuilder;
 
 class TimeoutWriter
 {
-    public static void WriteTimeoutScript(string scriptPath, BuildSqlDialect sqlDialect)
+    public static void WriteTimeoutScript(bool cleanup, bool overwrite, string scriptPath, BuildSqlDialect sqlDialect)
     {
         var createPath = Path.Combine(scriptPath, "Timeout_Create.sql");
-        File.Delete(createPath);
+        if (cleanup)
+        {
+            File.Delete(createPath);
+        }
+
+        CheckExistingFile(overwrite, createPath);
         using (var writer = File.CreateText(createPath))
         {
             TimeoutScriptBuilder.BuildCreateScript(writer, sqlDialect);
         }
+
         var dropPath = Path.Combine(scriptPath, "Timeout_Drop.sql");
-        File.Delete(dropPath);
+        if (cleanup)
+        {
+            File.Delete(dropPath);
+        }
+
+        CheckExistingFile(overwrite, dropPath);
         using (var writer = File.CreateText(dropPath))
         {
             TimeoutScriptBuilder.BuildDropScript(writer, sqlDialect);
         }
     }
+
+    static void CheckExistingFile(bool overwrite, string sagaFileName)
```

Comment:
Should be `timeoutFileName` or maybe just `fileName`?

Reply:
won't be applicable after the refactoring to the base class.